open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/common/idempotency/IdempotencyValidator.java

```diff
@@ -80,123 +79,146 @@ public static IdempotencyValidationResult validateIdempotency(String idempotency
-                                // Check whether difference between two dates is less than the configured time
-                                if (isRequestReceivedWithinAllowedTime(consentRequest.getCreatedTime())) {
-                                    // Compare whether JSON payloads are equal
-                                    if (isJSONPayloadSimilar(consentRequest.getReceipt(), request)) {
-                                        log.debug("Payloads are similar and request received within allowed time." +
-                                                " Hence this is a valid idempotent request");
-                                        return new IdempotencyValidationResult(true, true,
-                                                consentRequest, consentId);
-                                    } else {
-                                        log.debug(IdempotencyConstants.ERROR_PAYLOAD_NOT_SIMILAR);
-                                        throw new ConsentManagementException(IdempotencyConstants
-                                                .ERROR_PAYLOAD_NOT_SIMILAR);
-                                    }
-                                } else {
-                                    log.debug(IdempotencyConstants.ERROR_AFTER_ALLOWED_TIME);
-                                    throw new ConsentManagementException(IdempotencyConstants
-                                            .ERROR_AFTER_ALLOWED_TIME);
-                                }
-                            } else {
-                                log.debug(IdempotencyConstants.ERROR_MISMATCHING_CLIENT_ID);
-                                throw new ConsentManagementException(IdempotencyConstants.ERROR_MISMATCHING_CLIENT_ID);
-                            }
+                            return validateIdempotencyConditions(consentManageData, consentRequest);
                         } else {
-                            log.debug(IdempotencyConstants.ERROR_NO_CONSENT_DETAILS);
-                            throw new ConsentManagementException(IdempotencyConstants.ERROR_NO_CONSENT_DETAILS);
+                            log.error(IdempotencyConstants.ERROR_NO_CONSENT_DETAILS);
+                            throw new IdempotencyValidationException(IdempotencyConstants.ERROR_NO_CONSENT_DETAILS);
                         }
                     }
                 }
             } catch (IOException e) {
                 log.error(IdempotencyConstants.JSON_COMPARING_ERROR, e);
-                throw new ConsentManagementException(IdempotencyConstants.JSON_COMPARING_ERROR);
+                throw new IdempotencyValidationException(IdempotencyConstants.JSON_COMPARING_ERROR);
+            } catch (ConsentManagementException e) {
+                log.error(IdempotencyConstants.CONSENT_RETRIEVAL_ERROR, e);
+                return new IdempotencyValidationResult(true, false, null, null);
```

Comment:
Are the last two parameters of IdempotencyValidationResult constructor not applicable when one of the first two parameters are false? If that is the case, can't we have a separate constructor which takes only the first two params?

Reply:
Fixed in https://github.com/wso2/financial-open-banking/pull/51/commits/e1205691228fc4b76853a0eab7dd99dac2516661